### PR TITLE
refactor: pool/tick_math.gno

### PR DESCRIPTION
**Pull request by notJoon**
## Description

1. Extract tick ratio values into map (`tickRatioMap`)
2. Extracted the individual logic of the `TickMathGetTickAtSqrtRatio` function into functions.

### CODE REVIEW

File: pool/tick_math.gno [hunk 1]
```diff
@@ -20,143 +49,109 @@ func TickMathGetSqrtRatioAtTick(tick int32) bigint {
...
+// getTickValue determines the tick value corresponding to a given sqrtPriveX96.
+//
+// It calculates the upper and lower bounds for each tick, and selects the appropriate tock value
+// based on the given sqrtPriceX96.
+func getTickValue(log2, sqrtPriceX96 bigint) int32 {
+	// ref: https://github.com/Uniswap/v3-core/issues/500
+	// 2^64 / log2 (√1.0001) = 255738958999603826347141
+	log_sqrt10001 := log2 * 255738958999603826347141
 
+	// 0.010000497 x 2^128 = 3402992956809132418596140100660247210
```
Comment (r3v4s) [inline pool/tick_math.gno:151]: where did it came from `0.010000497` ?

File: pool/tick_math.gno [hunk 2]
```diff
@@ -20,143 +49,109 @@ func TickMathGetSqrtRatioAtTick(tick int32) bigint {
...
 	tickLow := int32(int64((log_sqrt10001 - 3402992956809132418596140100660247210) >> 128))
+
+	// 0.856 x 2^128 = 291339464771989622907027621153398088495
```
Comment (r3v4s) [inline pool/tick_math.gno:154]: and this `0.856`

**Review by r3v4s — COMMENTED** [review]
logic & tc looks good, just couldn't find where number in commnets came from.

@mconcat if pr is suitable for your uint256 approve it

**Comment by notJoon** [comment]
> couldn't find where number in commnets came from.

I forgot the source, please see the comments at that [link](https://ethereum.stackexchange.com/a/113912)

**Comment by r3v4s** [comment]
> > couldn't find where number in commnets came from.
> 
> @r3v4s I forgot the source, please see the comments at that [link](https://ethereum.stackexchange.com/a/113912)

hm... accuracy is 2^14 for some reason?? interesting...

File: pool/_TEST_tick_math_test.gno [hunk 3]
```diff
@@ -0,0 +1,65 @@
+package pool
+
+import (
+	"testing"
+)
+
+func TestFindMSB(t *testing.T) {
```
Comment (mconcat) [inline pool/_TEST_tick_math_test.gno:7]: Thanks for writing the test!

File: pool/tick_math.gno [hunk 4]
```diff
@@ -20,143 +49,111 @@ func TickMathGetSqrtRatioAtTick(tick int32) bigint {
...
-	if absTick&0x20000 != 0 {
-		ratio = (ratio * 0x5d6af8dedb81196699c329225ee604) >> 128
-	}
-	if absTick&0x40000 != 0 {
-		ratio = (ratio * 0x2216e584f5fa1ea926041bedfe98) >> 128
-	}
-	if absTick&0x80000 != 0 {
-		ratio = (ratio * 0x48a170391f7dc42444e8fa2) >> 128
+	ratio := consts.Q128
+	for mask, value := range tickRatioMap {
```
Comment (mconcat) [inline pool/tick_math.gno:53]: 👍 👍

File: pool/tick_math.gno [hunk 5]
```diff
@@ -5,13 +5,42 @@ import (
...
+	0x2000:  0xa9f746462d870fdf8a65dc1f90e061e5,
+	0x4000:  0x70d869a156d2a1b890bb3df62baf32f7,
+	0x8000:  0x31be135f97d08fd981231505542fcfa6,
+	0x10000: 0x9aa508b5b7a84e1c677de54f3e99bc9,
+	0x20000: 0x5d6af8dedb81196699c329225ee604,
+	0x40000: 0x2216e584f5fa1ea926041bedfe98,
+	0x80000: 0x48a170391f7dc42444e8fa2,
+}
+
+var binaryLogConsts = [8]bigint{
```
Comment (mconcat) [inline pool/tick_math.gno:31]: easier to read compared to `_tv`, 👍

**Review by mconcat — APPROVED** [review]

**Review by r3v4s — APPROVED** [review]

**Comment by r3v4s** [comment]
be aware of this pr in future, importing `r` in `p` will be forbidden
https://github.com/gnolang/gno/pull/1393

**Pull request merged**